From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: marcost2 <marcostolcachir@gmail.com>
Date: Sun, 22 Sep 2024 11:25:20 -0300
Subject: [PATCH] techpack: display: Make lhbm_not_allowed_fps_list_len signed

*count_u32_elems returns negative if property doesn't exist, this raises a stack trace while trying to read the array as the property does not exist

Original patch:
https://github.com/marcost2/android_kernel_motorola_sm6375/commit/418450761af43558fe2f733fdca9a51ede116938

Signed-off-by: Sevenrock <sevenrock@hotmail.de>
---
 techpack/display/msm/dsi/dsi_panel.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/techpack/display/msm/dsi/dsi_panel.h b/techpack/display/msm/dsi/dsi_panel.h
index c989468b1489..b0134670e403 100644
--- a/techpack/display/msm/dsi/dsi_panel.h
+++ b/techpack/display/msm/dsi/dsi_panel.h
@@ -249,7 +249,7 @@ struct dsi_panel_lhbm_config {
 	u32 lhbm_wait_for_fps_count;
 	u32 lhbm_wait_for_fps_interval;
 	u32 *lhbm_not_allowed_fps_list;
-	u32 lhbm_not_allowed_fps_list_len;
+	int lhbm_not_allowed_fps_list_len;
 };
 
 enum panel_idx {
-- 
2.39.5

